src/Symfony/Bundle/FrameworkBundle/Command/DotenvDebugCommand.php

```diff
@@ -0,0 +1,126 @@
+    private $projectDirectory;
+
+    public function __construct(string $kernelEnvironment, string $projectDirectory)
+    {
+        $this->kernelEnvironment = $kernelEnvironment;
+        $this->projectDirectory = $projectDirectory;
+
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+        $io->title('Dotenv Variables & Files');
+
+        $envFiles = $this->getEnvFiles();
+        $availableFiles = array_filter($envFiles, function (string $file) {
+            return is_file($this->projectDirectory.\DIRECTORY_SEPARATOR.$file);
+        });
+
+        if (\in_array('.env.local.php', $availableFiles, true)) {
+            $io->warning('Due to existing dump file (.env.local.php) all other dotenv files are skipped.');
+        }
+
+        $io->section('Available Files');
+        $io->listing(array_map(static function (string $envFile) use ($availableFiles) {
+            return \in_array($envFile, $availableFiles, true)
+                ? sprintf('<fg=green>✓</> %s', $envFile)
+                : sprintf('<fg=red>⨯</> %s', $envFile);
+        }, $envFiles));
+        $io->comment('Priority is descending');
+
+        $io->section('Variables with values');
+        $io->table(
+            array_merge(['Variable', 'Value'], $availableFiles),
+            $this->getVariables($availableFiles)
+        );
+
+        $io->comment('Note real values might be different between web and CLI.');
+
+        return 0;
+    }
+
+    private function getVariables(array $envFiles): array
+    {
+        $vars = [];
+        $fileValues = [];
+
+        foreach ($envFiles as $envFile) {
+            $file = $this->projectDirectory.\DIRECTORY_SEPARATOR.$envFile;
+            if (!is_file($file)) {
+                continue;
+            }
+            $values = $this->loadValues($file);
+            $vars = array_merge($vars, array_keys($values));
+            $fileValues[$envFile] = $values;
+        }
+        $vars = array_unique($vars);
+        sort($vars);
+
+        $output = [];
+        foreach ($vars as $var) {
+            $realValue = $_SERVER[$var];
+            $varDetails = [$var, $realValue];
+            foreach ($envFiles as $envFile) {
+                $varString = $fileValues[$envFile][$var] ?? '<fg=yellow>n/a</>';
+                $shortenedVar = mb_strlen($varString) > 18 ? substr($varString, 0, 15).'...' : $varString;
+                $varDetails[] = $varString === $realValue ? '<fg=green>'.$shortenedVar.'</>' : $shortenedVar;
+            }
+
+            $output[] = $varDetails;
+        }
+
+        return $output;
+    }
+
+    private function getEnvFiles(): array
+    {
+        return [
+            '.env.local.php',
+            sprintf('.env.%s.local', $this->kernelEnvironment),
+            sprintf('.env.%s', $this->kernelEnvironment),
+            '.env.local',
+            '.env',
```

Comment:
there's another case: https://github.com/symfony/symfony/blob/29b3d7bb73d56f69b54fc4e2c169a0e66b338d79/src/Symfony/Component/Dotenv/Dotenv.php#L99

Reply:
good catch! addressed in `getEnvFiles()`